src/net.h

```diff
@@ -1234,6 +1234,9 @@ class CNode
     void MaybeSetAddrName(const std::string& addrNameIn);
 
     std::string ConnectionTypeAsString() const;
+
+    /** Whether this peer is an inbound onion, e.g. connected via our Tor onion service. */
+    bool IsInboundOnion() const { return m_inbound_onion; }
```

Comment:
Given that this method is added only to be used in tests, could the tests use
`pnode->IsInboundConn() && pnode->ConnectedThroughNetwork() == NET_ONION` which is equivalent to the newly added `pnode->IsInboundOnion()`?

Reply:
The goal is to test `m_inbound_onion` to be able to depend on it in #20197.

Reply:
(the unit tests already assert on `IsInboundConn()`, so this would shift the coverage from `m_inbound_onion` directly, to indirectly via `ConnectedThroughNetwork() == NET_ONION`, which would avoid the getter but seems less robust?)

Reply:
Yeah, it will test it indirectly, less robust. Feel free to ditch this conversation :)